src/app/clients/getClientParcelsData.ts

```diff
@@ -0,0 +1,69 @@
+import supabase from "@/supabaseClient";
+import { DatabaseError } from "@/app/errorClasses";
+import { Data } from "@/components/DataViewer/DataViewer";
+import { formatDatetimeAsDate, formatDatetimeAsTime } from "@/app/parcels/getExpandedParcelDetails";
+
+export type RawClientParcelsDetails = Awaited<ReturnType<typeof getRawClientParcelsDetails>>;
+type ClientParcelDetails = RawClientParcelsDetails[number];
+interface ParcelsDetail {
+    parcel_id: string;
+    collection_centre?: { name: string; acronym: string } | null;
+    collection_datetime: string | Date | null;
+    packing_datetime: string | null;
+    voucher_number?: string | null;
+}
+export const getExpandedClientParcelsDetails = async (
+    clientId: string
+): Promise<ExpandedClientParcelDetails[]> => {
+    const rawClientParcelsDetails = await getRawClientParcelsDetails(clientId);
+    const formattedList = rawClientParcelsDetails.map((parcel) => {
+        return rawDataToExpandedClientParcelsDetails(parcel);
+    });
+    return formattedList;
+};
+const getRawClientParcelsDetails = async (clientId: string): Promise<ParcelsDetail[]> => {
+    const { data, error } = await supabase
+        .from("parcels")
+        .select(
+            `
+        parcel_id:primary_key,
+        
+        collection_centre:collection_centres ( 
+            name, 
+            acronym
+         ),
+         
+        collection_datetime,
+        packing_datetime,
+        voucher_number
+    `
+        )
+        .eq("client_id", clientId)
+        .order("packing_datetime", { ascending: false });
+
+    if (error) {
+        throw new DatabaseError("fetch", "parcel table data");
+    }
+
+    return data ?? [];
+};
+
+export interface ExpandedClientParcelDetails extends Data {
+    parcelId: string;
+    voucherNumber: string;
+    packingDate: string;
+    packingTime: string;
+    collectionCentre: string;
+}
+
+export const rawDataToExpandedClientParcelsDetails = (
```

Comment:
Should we write a unit test for this? You can have a look at `ParcelsPage.cy.tsx` for examples e.g. "Backend Processing for Expanded Parcel Details" tests.

We could probably make a new test file for `getClientParcelsData.ts` and name it `getClientParcelsData.cy.ts`? It'd nice to have a test file that corredponds to a source file.